container/lxd/container.go

```diff
@@ -262,11 +264,30 @@ func (s *Server) CreateContainerFromSpec(spec ContainerSpec) (*Container, error)
 			Profiles:     spec.Profiles,
 			Devices:      spec.Devices,
 			Config:       spec.Config,
-			Ephemeral:    false,
+			Ephemeral:    ephemeral,
 		},
 	}
 	op, err := s.CreateContainerFromImage(spec.Image.LXDServer, *spec.Image.Image, req)
 	if err != nil {
+		if IsLXDAlreadyExists(err) {
+			container, _, conErr := s.GetContainer(spec.Name)
+			if conErr != nil {
+				return nil, errors.Trace(conErr)
+			}
+
+			// If we don't match the spec from the container, then we're not
+			// sure what we've got here. Return the original error message.
+			if container.Architecture != spec.Architecture ||
```

Comment:
Should we also check that the container is in the running/started state?

I know that this solution corresponds with that suggested by @jameinel  on Launchpad, but my concern is that if there is some issue with an aborted creation on the LXD side, we can represent success without a usable instance.

Reply:
There isn't a way to wait for the status to change, so we'll get the provisioner task to bounce it and retry to see if it's running. Alternatively, we could implement a wait timeout ?